src/WebJobs.Script.WebHost/Diagnostics/Sanitizer.cs

```diff
@@ -44,6 +61,7 @@ internal static string Sanitize(string input)
                     int credentialEnd = t.IndexOfAny(ValueTerminators, startIndex);
 
                     t = t.Substring(0, startIndex) + SecretReplacement + (credentialEnd != -1 ? t.Substring(credentialEnd) : string.Empty);
+                    inputWithPublicTokensHidden = inputWithPublicTokensHidden.Substring(0, startIndex) + SecretReplacement + (credentialEnd != -1 ? inputWithPublicTokensHidden.Substring(credentialEnd) : string.Empty);
```

Comment:
Is this needed? It looks like we return `t` -- so this line isn't really doing anything. It makes sense -- you're searching the `inputWithPublicTokensHidden` string but replacing them in `t` (the real one) to preserve the original text.

Reply:
Without this while loop would never end as inputWithPublicTokensHidden would always have tokens we need to replace

Reply:
Oh, I see it now. Thanks for the explanation :-)